fhnw-ph-schreibberatung.csl

```diff
@@ -0,0 +1,476 @@
+    </names>
+  </macro>
+
+  <macro name="contributors">
+    <names variable="author">
+      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter="; " delimiter-precedes-last="never"/>
+      <label form="verb-short" prefix=" " suffix="." text-case="lowercase" strip-periods="true"/>
+      <substitute>
+        <text macro="editor"/>
+        <text macro="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+    <text macro="recipient"/>
+  </macro>
+
+  <macro name="contributors-short">
+    <names variable="author">
+      <name form="short" delimiter="/"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+  </macro>
+
+  <macro name="interviewer">
+    <names variable="interviewer" delimiter=", ">
+      <label form="verb" prefix=" " text-case="capitalize-first" suffix=" "/>
+      <name and="text" delimiter=", "/>
+    </names>
+  </macro>
+
+  <macro name="archive">
+    <group delimiter=". ">
+      <text variable="archive_location" text-case="capitalize-first"/>
+      <text variable="archive"/>
+      <text variable="archive-place"/>
+    </group>
+  </macro>
+
+  <macro name="access">
+    <group delimiter=". ">
+      <choose>
+        <if type="graphic report" match="any">
+          <text macro="archive"/>
+        </if>
+        <else-if type="bill book graphic legal_case motion_picture report song article-journal article-magazine article-newspaper thesis chapter paper-conference" match="none">
+          <text macro="archive"/>
+        </else-if>
+      </choose>
+      <text variable="DOI" prefix="doi:"/>
+      <choose>
+        <if type="legal_case" match="none">
+          <group prefix="[" suffix="]" delimiter="">
+            <text variable="URL" prefix="" suffix=""/>
+            <date variable="accessed" prefix="; " suffix="">
+              <date-part name="day" suffix="."/>
+              <date-part  form="numeric" name="month" suffix="."/>
+              <date-part name="year"/>
+            </date>
+          </group>
+        </if>
+      </choose>
+    </group>
+  </macro>
+
+  <macro name="title">
+    <choose>
+      <if variable="title" match="none">
+        <choose>
+          <if type="personal_communication" match="none">
+            <text variable="genre" text-case="capitalize-first"/>
+          </if>
+        </choose>
+      </if>
+      <else-if type="bill book graphic legal_case motion_picture report song broadcast" match="any">
+        <text variable="title" font-style="italic"/>
+      </else-if>
+      <else-if type="entry entry-dictionary entry-encyclopedia">
+        <text variable="title" font-style="normal"/>
+      </else-if>
+      <else>
+        <text variable="title" font-style="italic"/>
+      </else>
+
+    </choose>
+  </macro>
+
+  <macro name="edition">
+    <text variable="edition" suffix=" Aufl."/>
```

Comment:
normally what we do here is to test whether the variable is numeric and only then add Aufl/Edition. You can see this in many existing styles. 
Otherwise we display the field as is (since it might be saying something like "Vom Autor durchgesehene Auflage" already.)